File: src/main/java/com/example/librarymanagement/service/UserService.java
package com.example.librarymanagement.service;

import com.example.librarymanagement.model.entity.Journal;
import com.example.librarymanagement.model.entity.Reservation;
import com.example.librarymanagement.model.entity.User;

import java.util.List;

public interface UserService {
    User getUserByPhoneNumberOrEmail(String email, String phoneNumber);
    User getUserById(Long id);
    User createUser(User user);
    User updateUser(Long id, User updatedUser);
    List<User> findAll();
    List<Reservation> findReservationsByUser(Long userId);
    List<Journal> findJournalsByUser(Long userId);
    List<Journal> borrowBookFromLibrary(Long userId, Long libraryId, Long bookId);
    List<Reservation> reserveBookInLibrary(Long userId, Long libraryId, Long bookId);
    void cancelReservationInLibrary(Long userId, Long bookId);
    List<Journal> returnBookToLibrary(Long userId, Long libraryId, Long bookId);
    void deleteUser(Long id);
}

File: src/main/java/com/example/librarymanagement/service/UserService.kt
package com.example.librarymanagement.service

import com.example.librarymanagement.model.domain.Journal
import com.example.librarymanagement.model.domain.Reservation
import com.example.librarymanagement.model.domain.User

interface UserService {
    fun getUserByPhoneNumberOrEmail(email: String?, phoneNumber: String?): User
    fun getUserById(id: String): User
    fun createUser(user: User): User
    fun updateUser(updatedUser: User): User
    fun findAll(): List<User>
    fun findJournalsByUser(userId: String): List<Journal>
//    fun deleteUserById(id: String)
}
